# Projeto Multidisciplinar I do curso Formação Plena em Ciência de Dados [TEORIA + PRÁTICA 2023]

Projeto Multidisciplinar I da aula 23, presente no curso Formação Plena em Ciência de Dados [TEORIA + PRÁTICA 2023] disponível na plataforma Udemy.

## Apresentação e discussão

O Projeto Multidisciplinar I consiste na simulação de dados envolvendo pesos e alturas de indivíduos, calculando o IMC (índice de Massa Corporal) e relacionando com dados simulados sobre existência de problemas cardíacos. A simulação deverá contar com dados pseudoaleatórios gerados por simulação computacional e deverá haver correspondência entre problemas cardiácos com a medida do IMC. Ou seja, iremos supor que exista uma correspondência óbvia entre as medidas (modelada matematicamente por equção e coeificentes arbitrariamente definidos). Para cada sexo (sendo eles definidos de forma aleatória).

Você, portanto, simulará via randomização (aleatoriedade) as alturas dentro de uma faixa bem definida. Em seguida, aplicará uma equação que determine, com algum grau de aleatoriedade, os pesos dos indivíduos. Logo depois, calculará o IMC e definirá para cada indivíduo a existência (sim ou não) de problemas cardiácos, ambos com algum grau mnimo de correspondência com o IMC.

Ao final, faça as análises necessárias com medidas de dispersão e de tendência central. Crie as funções para tal. E interprete os IMCs obtido conforme WHO (Organização Mundial da Saúde).

Os dados devem estar reunidos em uma tabela. Todas as técnicas aplicadas aqui são vistas na Aula 22.

## Gerando os dados simulados

In [3]:
n = 3000 #tamanho da amostra

In [4]:
#Gerando dados de média 1.73 e desvio padrão 0.30, sendo 3000 pessoas em uma distribuição normalizada.

import numpy
altura = numpy.random.normal(1.73, 0.30, n)
altura = altura.tolist()

In [5]:
#Escrevendo a função a ser usada para gerar os pesos
from sympy import *
coeficiente_g = symbols("a") #coeficiente a
var_1 = symbols("x") #variavel x da função linear
const_1 = symbols("c") #constante da função
peso_i = coeficiente_g*var_1+const_1
#const_1 varia no intervalo de -5 a 5 e segue distriuição uniforme conforme exemplo do instrutor
#coeficiente_g será gerado aleatoriamente entre 47 e 51 para tentativa de maior diversificação nos valores dos pesos
peso_i

a*x + c

In [6]:
from random import *
pesos = []

for i in altura:
    a = round(float(uniform(-5, 5)), 2)
    b = i*numpy.random.normal(49, 2)
    c = a + b
    pesos.append(c)

In [7]:
#formula do imc
peso1, altura1 = symbols(['peso', 'altura'])
IMC_formula = peso1 / (altura1**2)
IMC_formula

peso/altura**2

In [8]:
IMC = []

for i,j in zip(altura, pesos):
    e = j / (i**2)
    e = round(e, 2)
    IMC.append(e)

In [9]:
SEXO = []

for i in range(n):
    valor = numpy.random.uniform(0,1)
    if valor > 0.5:
        SEXO.append("F")
    else:
        SEXO.append("M")

## Fazendo as associações

In [10]:
CARDIACO = []

for i in range(n):
    if IMC[i] > 39.9:
        CARDIACO.append("S")
    elif IMC[i] > 30 and IMC[i] < 39.9:
        CARDIACO.append("T")
    else:
        CARDIACO.append("N")

In [11]:
tabela = altura, pesos, IMC, SEXO, CARDIACO

In [12]:
import pandas
df = pandas.DataFrame(tabela, index=["Altura", "Peso", "IMC", "Sexo", "Cardiaco"]).T

In [13]:
df

Altura        Peso    IMC Sexo Cardiaco
0     2.007505  101.995142  25.31    M        N
1     1.985197   98.015834  24.87    F        N
2     1.513134   74.170006  32.39    M        T
3     1.502914   75.002317  33.21    F        T
4     2.286443  107.933676  20.65    M        N
...        ...         ...    ...  ...      ...
2995   2.16359  103.226061  22.05    M        N
2996   1.73916   76.048119  25.14    F        N
2997  2.103869   98.343258  22.22    M        N
2998  1.985689   93.399716  23.69    M        N
2999  1.912684   90.182216  24.65    F        N

[3000 rows x 5 columns]

## Medidas de tendência central

In [36]:
#Calculando a média de uma coluna na tabela
def media(tabel):
    soma = sum(tabel)
    media = round(soma/len(tabel), 2)
    return media

In [37]:
print(media(df.Altura))
print(media(df.Peso))
print(media(df.IMC))

1.73
84.57
29.34


In [16]:
#calculando a mediana de uma coluna na tabela

def mediana(tabela):
    tabela = sorted(tabela)
    index = (len(tabela)-1)//2
    if len(tabela)%2:
        print(tabela[index])
    else:
        print((tabela[index]+tabela[index+1])/2)

mediana(df.Altura)
mediana(df.Peso)
mediana(df.IMC)

1.7217309977945146
84.22188617915427
28.395


In [17]:
#media para 50% inferior de uma coluna

def fiftyinf(tabela):
    return numpy.mean(tabela.sort_values(ascending=True)[0:1500])

print(fiftyinf(df.Altura))
print(fiftyinf(df.Peso))


1.4909450116395262
72.53888095691612
25.021853333333333


In [50]:
#media para 50% superior de uma coluna

def fiftysup(tabela):
    return numpy.mean(tabela.sort_values(ascending=True)[1500:n])

print(fiftysup(df.Altura))
print(fiftysup(df.Peso))

1.9594262009459222
96.61028316278372
33.652573333333294


In [49]:
#media do peso para 50% pessoas mais baixas

#index = (len(tabela)-1)//2
dfMenoresAlturas = df.sort_values(ascending=True, by=["Altura"])[0:1500]
numpy.mean(dfMenoresAlturas.IMC)

33.457353333333394

In [48]:
#media do peso para 50% pessoas mais altas

dfMaioresAlturas = dfOrdenadoAlturas[1500:n]
numpy.mean(dfMaioresAlturas.IMC)

25.217073333333314

In [42]:
#variancia e desvio padrão de uma coluna

def variancia_std(tabela):
    variancia = 0
    media_coluna = media(tabela)
    for valor in tabela:
        variancia += (valor - media_coluna)**2
    variancia = variancia / (n - 1)
    desvio = sqrt(variancia)
    print(f'Variância: {variancia}, Desvio Padrão: {desvio}')

variancia_std(df.Altura)

Variância: 0.08652994269790386, Desvio Padrão: 0.294159723106179


##  Frequência para valores qualitativos

In [22]:
#Frequência dos sexos
freq_m = 0
freq_f = 0
for i in df.Sexo:
    if i == "M":
        freq_m += 1
    else:
        freq_f += 1
print("O percentual de homens é:", round((freq_m/n*100),2))
print("O percentual de mulheres é:", round((freq_f/n*100),2))

O percentual de homens é: 51.37
O percentual de mulheres é: 48.63


In [23]:
#Frequência tipos de cardíacos
freq_s = 0
freq_t = 0
freq_n = 0

for i in df.Cardiaco:
    if i == "S":
        freq_s += 1
    else:
        if i == "T":
            freq_t += 1
        else:
            freq_n += 1
print("O percentual de pessoas com problemas cardíacos é:", round((freq_s/n*100),2))
print("O percentual de pessoas com tendência a problemas cardiacos é:", round((freq_t/n*100),2))
print("O percentual de pessoas sem tendência a problemas cardiacos é:", round((freq_n/n*100),2))
freq_s, freq_t, freq_n

O percentual de pessoas com problemas cardíacos é: 5.27
O percentual de pessoas com tendência a problemas cardiacos é: 32.83
O percentual de pessoas sem tendência a problemas cardiacos é: 61.9


(158, 985, 1857)

## Análise final

    População com estatura média normal (1,73) e com peso médio elevado para a altura. IMC médio em nível de pré-obesidade. 51,37% são do sexo masculino enquanto 48.63% é do sexo feminino. Em geral, 5,27% das pessoas possuem problemas cardíacos e 32,83% possuem tendência a problemas cardíacos. Pode-se observar que as pessoas mais baixas, em média, possuem um IMC mais baixo. Maiores análises podem ser feitas para relacionar a média de IMC e de problemas cardíacos categorizado por sexo.